openedx/features/course_bookmarks/static/course_bookmarks/js/spec/bookmarks_list_view_spec.js

```diff
@@ -0,0 +1,260 @@
+             BookmarkHelpers, BookmarksListView, BookmarksCollection) {
+        'use strict';
+
+        describe('BookmarksListView', function() {
+            var createBookmarksView, verifyRequestParams;
+
+            beforeEach(function() {
+                loadFixtures('course_bookmarks/fixtures/bookmarks.html');
+                TemplateHelpers.installTemplates([
+                    'templates/fields/message_banner'
+                ]);
+                spyOn(Logger, 'log').and.returnValue($.Deferred().resolve());
+                jasmine.addMatchers({
+                    toHaveBeenCalledWithUrl: function() {
+                        return {
+                            compare: function(actual, expectedUrl) {
+                                return {
+                                    pass: expectedUrl === actual.calls.mostRecent().args[0].currentTarget.pathname
+                                };
+                            }
+                        };
+                    }
+                });
+            });
+
+            createBookmarksView = function() {
+                var bookmarksCollection = new BookmarksCollection(
+                    [],
+                    {
+                        course_id: BookmarkHelpers.TEST_COURSE_ID,
+                        url: BookmarkHelpers.TEST_API_URL
+                    }
+                );
+                var bookmarksView = new BookmarksListView({
+                    $el: $('.course-bookmarks'),
+                    collection: bookmarksCollection,
+                    loadingMessageView: new MessageBannerView({el: $('#loading-message')}),
+                    errorMessageView: new MessageBannerView({el: $('#error-message')})
+                });
+                return bookmarksView;
+            };
+
+            verifyRequestParams = function(requests, params) {
+                var urlParams = (new URI(requests[requests.length - 1].url)).query(true);
+                _.each(params, function(value, key) {
+                    expect(urlParams[key]).toBe(value);
+                });
+            };
+
+            it('can correctly render an empty bookmarks list', function() {
+                var requests = AjaxHelpers.requests(this);
+                var bookmarksView = createBookmarksView();
+                var expectedData = BookmarkHelpers.createBookmarksData({numBookmarksToCreate: 0});
+
+                bookmarksView.showBookmarks();
+                AjaxHelpers.respondWithJson(requests, expectedData);
+
+                expect(bookmarksView.$('.bookmarks-empty-header').text().trim()).toBe(
+                    'You have not bookmarked any courseware pages yet'
+                );
+
+                expect(bookmarksView.$('.bookmarks-empty-detail-title').text().trim()).toBe(
+                    'Use bookmarks to help you easily return to courseware pages. ' +
+                    'To bookmark a page, click on "Bookmark this page" beneath the unit title.'
```

Comment:
@andy-armstrong My suggestions: strike "on" and use "under" instead of "beneath".
"To bookmark a page, click "Bookmark this page" under the unit title."